pkg/application/registryConfigUseCase_test.go

```diff
@@ -0,0 +1,651 @@
+		expectedSpec.Events = append(spec.Events, domain.RegistryConfigAppliedEvent{})
+		blueprintRepoMock := newMockBlueprintSpecRepository(t)
+
+		blueprintRepoMock.EXPECT().Update(testCtx, expectedSpec).Return(nil)
+
+		sut := EcosystemRegistryUseCase{blueprintRepository: blueprintRepoMock}
+
+		// when
+		err := sut.markConfigApplied(testCtx, spec)
+
+		// then
+		require.NoError(t, err)
+	})
+
+	t.Run("should return an error on update error", func(t *testing.T) {
+		// given
+		spec := &domain.BlueprintSpec{}
+		expectedSpec := &domain.BlueprintSpec{}
+		expectedSpec.Status = domain.StatusPhaseRegistryConfigApplied
+		expectedSpec.Events = append(spec.Events, domain.RegistryConfigAppliedEvent{})
+		blueprintRepoMock := newMockBlueprintSpecRepository(t)
+
+		blueprintRepoMock.EXPECT().Update(testCtx, expectedSpec).Return(assert.AnError)
+
+		sut := EcosystemRegistryUseCase{blueprintRepository: blueprintRepoMock}
+
+		// when
+		err := sut.markConfigApplied(testCtx, spec)
+
+		// then
+		require.Error(t, err)
+		assert.ErrorContains(t, err, "failed to mark registry config applied")
+		assert.ErrorIs(t, err, assert.AnError)
+	})
+}
+
+func TestEcosystemRegistryUseCase_markApplyConfigStart(t *testing.T) {
+	t.Run("should set status and event apply config", func(t *testing.T) {
+		// given
+		spec := &domain.BlueprintSpec{}
+		expectedSpec := &domain.BlueprintSpec{}
+		expectedSpec.Status = domain.StatusPhaseApplyRegistryConfig
+		expectedSpec.Events = append(spec.Events, domain.ApplyRegistryConfigEvent{})
+		blueprintRepoMock := newMockBlueprintSpecRepository(t)
+
+		blueprintRepoMock.EXPECT().Update(testCtx, expectedSpec).Return(nil)
+
+		sut := EcosystemRegistryUseCase{blueprintRepository: blueprintRepoMock}
+
+		// when
+		err := sut.markApplyConfigStart(testCtx, spec)
+
+		// then
+		require.NoError(t, err)
+	})
+
+	t.Run("should return an error on update error", func(t *testing.T) {
+		// given
+		spec := &domain.BlueprintSpec{}
+		expectedSpec := &domain.BlueprintSpec{}
+		expectedSpec.Status = domain.StatusPhaseApplyRegistryConfig
+		expectedSpec.Events = append(spec.Events, domain.ApplyRegistryConfigEvent{})
+		blueprintRepoMock := newMockBlueprintSpecRepository(t)
+
+		blueprintRepoMock.EXPECT().Update(testCtx, expectedSpec).Return(assert.AnError)
+
+		sut := EcosystemRegistryUseCase{blueprintRepository: blueprintRepoMock}
+
+		// when
+		err := sut.markApplyConfigStart(testCtx, spec)
+
+		// then
+		require.Error(t, err)
+		assert.ErrorContains(t, err, "cannot mark blueprint as applying config")
+		assert.ErrorIs(t, err, assert.AnError)
+	})
+}
+
+func TestEcosystemRegistryUseCase_handleFailedApplyRegistryConfig(t *testing.T) {
+	t.Run("should set applied status and event", func(t *testing.T) {
+		// given
+		spec := &domain.BlueprintSpec{}
+		blueprintRepoMock := newMockBlueprintSpecRepository(t)
+
+		blueprintRepoMock.EXPECT().Update(testCtx, mock.IsType(&domain.BlueprintSpec{})).Return(nil)
```

Comment:
Why is this test written unlike the others? The other ones used an ExpectedSpec and checked in the mock-call. I think they should be similiar.

Reply:
Because you can't set an error in the event here. The property is not public and in this package.